### PR TITLE
Parallelize nightwatch sitemap a11y tests 

### PR DESCRIPTION
**Pull request by billfienberg**
## GitHub Issue

https://github.com/department-of-veterans-affairs/va.gov-team/issues/22445

## Description

This PR does the following:

- Updates the parallelization config ([`test_workers`](https://nightwatchjs.org/gettingstarted/configuration/#test-runner-settings)) of the Nightwatch Sitemap a11y tests from `false` to `true`, and specifies `4` workers because the sitemap tests are chunked into four separate files (`src/platform/site-wide/tests/sitemap/sitemap-[1-4].spec.js`). 

Caveat: The `nightwatch-accessibility` step only runs on the `master` branch, so this might speed up CI runs **after** a branch is merged to `master` and deploys, but it won't have any effect on pull requests. 😞  

## Testing done

CI

### [Before](http://jenkins.vfs.va.gov/blue/organizations/jenkins/testing%2Fvets-website/detail/master/10705/pipeline/152/): 1:05:15

image

### [After](http://jenkins.vfs.va.gov/blue/organizations/jenkins/testing%2Fvets-website/detail/parallelize-nightwatch-sitemap-a11y-tests/9/pipeline/110/): 20:46

image

## Methodology

Since the `nightwatch-accessibility` step only runs against the `PROD_BRANCH` (`master`), I temporarily added the `nightwatch-accessibility` step to run against my branch. Once I got a screenshot of the time it took to run against my branch, I deleted that temporary code. 

I got the baseline time AKA the before time from a [recent passing run on `master`](http://jenkins.vfs.va.gov/blue/organizations/jenkins/testing%2Fvets-website/detail/master/10705/pipeline/152/). 

## Acceptance criteria
- [ ] The `nightwatch-accessibility` step takes less than an hour to finish

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

## Links

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/22443
- https://nightwatchjs.org/gettingstarted/configuration/#test-runner-settings
- https://markus.oberlehner.net/blog/speeding-up-nightwatch-powered-acceptance-tests/

### CODE REVIEW

**Review by U-DON — APPROVED** [review]
This is great! Nicely done. 🚀

File: src/platform/site-wide/tests/sitemap/sitemap-helpers.js [hunk 1]
```diff
@@ -56,18 +56,17 @@ function sitemapURLs() {
...
 function runTests(client, segment, only508List) {
   segment.forEach(url => {
+    console.time(url); // eslint-disable-line no-console
     const only508 = only508List.filter(path => url.endsWith(path)).length > 0;
     client
-      .perform(() => {
-        // eslint-disable-next-line no-console
-        console.log(url);
-      })
+      .perform(() => {})
```
Comment (billfienberg) [inline src/platform/site-wide/tests/sitemap/sitemap-helpers.js:62]: I imagine this line can be deleted, but I didn't feel like restarting CI to find out.

**Comment by pjhill** [comment]
Epic!
Bonus points for also cutting the Nighwatch-E2E test from --
[Before: 5m 42s](http://jenkins.vfs.va.gov/blue/organizations/jenkins/testing%2Fvets-website/detail/master/10709/pipeline/151)
image
[After: 2m 32s](http://jenkins.vfs.va.gov/blue/organizations/jenkins/testing%2Fvets-website/detail/master/10711/pipeline/151)
image

**Comment by ncksllvn** [comment]
This is awesome @billfienberg  - just wondering, did you try intentionally creating an a11y error in a template to ensure that CI will still fail when it runs into one?

**Comment by billfienberg** [comment]
> This is awesome - just wondering, did you try intentionally creating an a11y error in a template to ensure that CI will still fail when it runs into one?

@ncksllvn, great question. 

Since you asked, I created a [branch](https://github.com/department-of-veterans-affairs/vets-website/compare/22445-test-the-test) where I’m testing my test, and I can’t seem to get my nightwatch sitemap tests to fail.

I’ve tried the following:
- Adding a serious error (label-less input) to src/applications/facility-locator/containers/FacilityLocatorApp.jsx (5c39f88)
- Adding a critical error (Buttons must have discernible text) to src/applications/edu-benefits/feedback-tool/containers/IntroductionPage.jsx (a7d147b)
- Adding a critical error (Buttons must have discernible text) to a liquid template (ba66048)

I'm currently discussing this with Trevor in a Slack thread.

**Comment by billfienberg** [comment]
Update from Trevor:

>I removed a custom middlware and made it simpler to pass in config object overrides like we did for Cypress. The middleware was always returning an empty violations array, quietly.
>
>Once teams get all the upstream axe violations cleaned out, I'll merge my [PR](https://github.com/department-of-veterans-affairs/vets-website/pull/16424) and fixes for a handful of Nightwatch e2e tests and it'll be 🦄   and 🌈  again.

**Pull request merged**